src/webapp/components/dataset-periods/DataSetPeriodDates.tsx

```diff
@@ -0,0 +1,186 @@
+    };
+
+    const onUpdatePeriodDate = (date: string, fieldName: "startDate" | "endDate") => {
+        const newValues = periodDate?.setDates(date, fieldName);
+        setPeriodDate(newValues);
+    };
+
+    return (
+        <ConfirmationDialog
+            cancelText={i18n.t("Close")}
+            fullWidth
+            onCancel={onCancel}
+            open
+            title={i18n.t("Set output/outcome period dates: {{dataSetsNames}}", {
+                nsSeparator: true,
+                dataSetsNames: dataSetNames,
+            })}
+            saveText={i18n.t("Save")}
+            onSave={() => onSavePeriods()}
+            disableSave={!periodDate.startDate || !periodDate.endDate}
+        >
+            <DatesContainer>
+                <DatePicker
+                    value={periodDate.startDate || null}
+                    onChange={value => onUpdatePeriodDate(value, "startDate")}
+                    label={i18n.t("Start date of data input")}
+                    format="yyyy-MM-DD"
+                />
+                <DatePicker
+                    value={periodDate.endDate || null}
+                    onChange={value => onUpdatePeriodDate(value, "endDate")}
+                    label={i18n.t("End date of data input")}
+                    minDate={periodDate.startDate || ""}
+                    format="yyyy-MM-DD"
+                />
+            </DatesContainer>
+            {periodsByYear.length > 0 && (
+                <PeriodsContainer>
+                    <Typography className="periods-text">{i18n.t("Periods")}</Typography>
+                    {periodsByYear.map(periodDetail => {
+                        return (
+                            <PeriodDateItem key={periodDetail.year}>
+                                <span>
+                                    <strong>{periodDetail.year}</strong>
+                                </span>
+                                <DatePicker
+                                    value={periodDetail.startDate}
+                                    onChange={value =>
+                                        updatePeriod(periodDetail, value, "startDate")
+                                    }
+                                    label={i18n.t("Start date of data input")}
+                                    className="datepicker"
+                                    minDate={periodDate.startDate}
+                                    format="yyyy-MM-DD"
+                                />
+                                <DatePicker
+                                    value={periodDetail.endDate}
+                                    onChange={value => updatePeriod(periodDetail, value, "endDate")}
+                                    label={i18n.t("End date of data input")}
+                                    minDate={periodDate.startDate}
+                                    className="datepicker"
+                                    format="yyyy-MM-DD"
```

Comment:
repeated format. Either a variable or a wrapper component that adds common props.